lib/_tls_common.js

```diff
@@ -52,6 +54,13 @@ function SecureContext(secureProtocol, secureOptions, context) {
   if (secureOptions) this.context.setOptions(secureOptions);
 }
 
+function validateKeyCert(value, type) {
+  if (typeof value !== 'string' && !(value instanceof Buffer))
```

Comment:
I think this is too strict, and `instanceof Buffer` is not a foolproof way to check for Buffers. Can you use `ArrayBuffer.isView(value)` instead and adjust the message accordingly (Buffers are just special Uint8Arrays, but all Uint8Arrays and other ArrayBufferViews should work, I think.).